examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java

```diff
@@ -60,9 +68,12 @@ class TableTab extends ScrollableTab {
 	static final int CELL_FONT = 8;
 	static final int HEADER_FOREGROUND_COLOR = 9;
 	static final int HEADER_BACKGROUND_COLOR = 10;
-	Color itemForegroundColor, itemBackgroundColor, cellForegroundColor, cellBackgroundColor, headerForegroundColor, headerBackgroundColor;
+	Color itemForegroundColor, itemBackgroundColor, cellForegroundColor, cellBackgroundColor, headerForegroundColor,
+			headerBackgroundColor;
 	Font itemFont, cellFont;
 
+	static int column = -1;
```

Comment:
Why is this static field necessary? The value does only seem to be used in the mouse click handler, but there is no need to store it across multiple invocations of that handler. In case there is reason for storing this state across multiple method invocations, why does it need to be static?

Reply:
@ShahzaibIbrahim I still do not see any need to this field. Can you please explain why it is necessary?

Reply:
@HeikoKlare We use this column field to identify which column is selected for editing and then add the text. See the line 307 to 311.

Reply:
But isn't that value calculated anew everytime the listener is called (in lines 294 to 300)? Then why store it in a field?

Reply:
My question is why the value needs to be stored outside the lambda at all. Why not simply define a variable `int column = -1` at the beginning of the lamda? What is the scenario in which the value stored in the previous execution of the lambda needs to be available in a subsequent execution of it?

Reply:
I tried that too but again I needed to use this field into another lambda (modify listener) inside (308 to 310) and I ran to the same problem again.

Reply:
For Context:

![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/49493944/65f9764b-e46f-436c-aa1e-a45f81a596cd)

Reply:
You can still assign the calculated value to a local final variable that is accessed from within the lambda.

To make it more clean, you can even extract the logic for finding the column containing the point into a separate method and assign the result to a local final variable, i.e., replace lines 294 to 300 with:
```java
final int column = findColumnContainingPoint(item, point);
```
and add an according method:
```java
	private static int findColumnContainingPoint(TableItem item, Point point) {
		for (int i = 0; i < item.getParent().getColumnCount(); i++) {
			Rectangle rect = item.getBounds(i);
			if (rect.contains(point)) {
				return i;
			}
		}
		return -1;
	}
```